sql/core/src/main/scala/org/apache/spark/sql/functions.scala

```diff
@@ -1931,6 +1931,42 @@ object functions extends LegacyFunctions {
     new Murmur3Hash(cols.map(_.expr))
   }
 
+  /**
+   * Encrypts input using AES and Returns the result as a binary column.
+   * Key lengths of 128, 192 or 256 bits can be used. 192 and 256 bits keys can be used if Java
+   * Cryptography Extension (JCE) Unlimited Strength Jurisdiction Policy Files are installed. If
+   * either argument is NULL, the result will also be null. If input is invalid, key length is not
+   * one of the permitted values or using 192/256 bits key before installing JCE, an exception will
+   * be thrown.
+   *
+   * @param input binary column to encrypt input
+   * @param key binary column of 128, 192 or 256 bits key
+   *
+   * @group misc_funcs
+   * @since 2.0.0
+   */
+  def aes_encrypt(input: Column, key: Column): Column = withExpr {
```

Comment:
then we should probably just take a string.

the thing is we want to facilitate the most common cases, e.g. if 99% of the time people are passing in keys directly rather than relying on some other columns' values, we should just let them pass the literals.